src/Components/Providers/ProductProvider.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useContext, createContext, useState, useEffect } from 'react';
+import axios from 'axios';
+
+const productContext = createContext(null);
+
+const ProductProvider = ({ children }) => {
+    const [ products, setProducts ] = useState([]);
+
+    useEffect(() => {
+        (
+            async () => {
```

Comment:
appreciation: good that you used IIFE function here!

Reply:
Thanks